gdsfactory/pdk.py

```diff
@@ -686,6 +686,17 @@ def to_updk(self) -> str:
     # def on_cross_section_registered(self) -> Event:
     #     return self._on_cross_section_registered
 
+    def get_cross_section_name(self, cross_section: CrossSection) -> str:
```

Comment:
in this case, we are getting the name of a specific cross section (i.e. rib of width 0.5 um) rather than the type of cross section, right? i suppose this should be ok, but just noting it's different from what i was doing before for pathlength extraction. it comes down to if you want to group all rib as rib or break them out based on particular widths, etc. ideally, i think you probably want to be able to do it either way

Reply:
the CrossSections are uniquely identified by name,

it will try to get name from the PDK, and if it does not exist it will name them with a hash

what do you think?